openedx/features/course_duration_limits/models.py

```diff
@@ -48,6 +47,36 @@ class CourseDurationLimitConfig(StackedConfigurationModel):
         )
     )
 
+    @classmethod
+    def has_full_access_role_in_masquerade(cls, user, course_key, course_masquerade):
+        """
+        When masquerading as a user, the course duration limits will never trigger the course to expire, redirecting the user.
+        However, the user masquerade roles are still used to determine whether the course duration limits banner will display.
+        Enable the course duration limits banner for the enrollment if a masquerading user has any of the following roles
```

Comment:
We actually are disabling the banner for these elevated roles.

Reply:
wrong comment good catch thanks